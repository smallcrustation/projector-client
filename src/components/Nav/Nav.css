.nav-bar a{
  padding: .5em 1em;
  margin: 0;
  box-shadow: none;
  border-bottom: none;
  border-top: none;
}

.nav-bar a:nth-child(even){
  border-left: none;
}


.nav-bar{
  display: flex;
  align-content: flex-end;
}
