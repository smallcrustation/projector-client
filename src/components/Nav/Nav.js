import React from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import {UserContext} from '../../contexts/UserContext'

import './Nav.css'

export default class Nav extends React.Component {
  static contextType = UserContext

  handleLogOut = () => {
    TokenService.clearAuthToken()
    this.context.setUser(null)
  }

  renderLoggedInNav() {
    // console.log('Nav State: ', this.state)
    return (
      <nav className="logged-in-nav nav-bar">
        <Link to="/projects">Projects</Link>
        <Link onClick={this.handleLogOut} to="/">
          Logout
        </Link>
      </nav>
    )
  }

  renderLoggedOutNav() {
    return (
      <nav className="logged-out-nav nav-bar">
        <Link to="/login">Login</Link>
        <Link to="/createAccount">Create</Link>
      </nav>
    )
  }

  render() {
    return (
      // <nav className="nav-bar">
        TokenService.getAuthToken()
          ? this.renderLoggedInNav()
          : this.renderLoggedOutNav()
      // </nav>
    )
  }
}
