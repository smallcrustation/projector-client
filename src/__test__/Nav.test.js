import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'

import Nav from '../components/Nav/Nav'
import renderer from 'react-test-renderer'

describe('App Testing', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
      div
    )

    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders with no crashing and matches snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
