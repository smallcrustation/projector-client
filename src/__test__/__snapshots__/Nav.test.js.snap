// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing Renders with no crashing and matches snapshot 1`] = `
<nav
  className="logged-out-nav nav-bar"
>
  <a
    className="link-btn"
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
  <a
    className="link-btn"
    href="/createAccount"
    onClick={[Function]}
  >
    Create
  </a>
</nav>
`;
